tests/refc/callingConvention/expected

```diff
@@ -10,38 +10,24 @@ Value *Main_last
 ,   Value * var_1
 )
 {
-    Value * tmp_131 = NULL;                                  // Main:5:8--5:14
-    AConAlt * constructorField_130= newConstructorField(2);  // Main:5:8--5:14
-    constructorField_130[0].tag = 0;
-    constructorField_130[0].name = NULL;
-    constructorField_130[1].tag = 1;
-    constructorField_130[1].name = NULL;
-    switch(compareConstructors(var_0, constructorField_130, 2)){
-                                                             // Main:5:8--5:14
-      case 0:
-      {
-        tmp_131 = newReference(var_1);
-        break;
-      }
-      case 1:
-      {
-        Value * var_2 = ((Value_Constructor*)var_0)->args[0];
-        Value * var_3 = ((Value_Constructor*)var_0)->args[1];
+    Value * tmp_128 = NULL;                                  // Main:5:8--5:14
+    if (NULL == var_0 /* Prelude.Basics.Nil [nil] */) {
+        tmp_128 = newReference(var_1);
+    } else if (NULL != var_0 /* Prelude.Basics.(::) [cons] */) {
```

Comment:
IIUC we are generating these redundant checks because we are emitting code in
such a way that each `MkAConAlt` is dealt with in isolation.

Is it not the case however that in the `NULL`-discrimination case the same mechanism
deciding whether to print an `else` could be used to decide to bypass the check?

(Side-remark: we should *really* move away from string-printing as the main mechanism
to generate C code. I have once again offered designing a C-- IR as an MSc project for
this year. Not sure whether there'll be any takers).

Reply:
I also thought this redundancy was a bit ugly. But modern C compilers can easily remove it. So I left the task to them. In other words, I simply ignored it.

In technically, it seems that it could be easily removed by simply sorting the case clause. For example, transform "[] => ... ;x::xs => ...." into "x::xs => ... ;_ => ..." .
On the other hand, such term simplification seems to be useful for all backends. In other words, I guess, this is not a backend's job, but something that should be solved a bit more upstream.

Reply:
Fair enough.